azurelinuxagent/ga/exthandlers.py

```diff
@@ -674,7 +674,7 @@ def launch_command(self, cmd, timeout=300):
             raise ExtensionError("Failed to launch: {0}, {1}".format(cmd, e))
 
         retry = timeout / 5
-        while retry > 0 and child.poll == None:
+        while retry > 0 and child.poll is None:
```

Comment:
You are correct in that this should be a method call, but it returns the returncode, not sets it. There are some issues with the tests as well though, so I will look into that next.